query/src/test/java/io/keen/client/java/KeenQueryTest.java

```diff
@@ -542,8 +559,1014 @@ public void testNullResult() throws Exception {
                 .build();
         mockCaptureCountQueryRequest(queryParams);
     }
+    
+    @Test
+    public void testFunnelWithOnlyRequiredParameters() throws Exception {
+        setMockResponse(200,
+            "{\"result\": [3,1,0],\"steps\":["
+          + "{\"actor_property\":\"visitor.guid\",\"event_collection\":\"signed up\",\"timeframe\":\"this_7_days\"},"
+          + "{\"actor_property\":\"user.guid\",\"event_collection\":\"completed profile\",\"timeframe\":\"this_7_days\"},"
+          + "{\"actor_property\":\"user.guid\",\"event_collection\":\"referred user\",\"timeframe\":\"this_7_days\"}]}");
+    
+        Funnel funnel = new Funnel.Builder()
+                .withStep(new FunnelStep("signed up", "visitor.guid", new RelativeTimeframe("this_7_days")))
+                .withStep(new FunnelStep("completed profile", "user.guid", new RelativeTimeframe("this_7_days")))
+                .withStep(new FunnelStep("referred user", "user.guid", new RelativeTimeframe("this_7_days", "UTC")))
+                .build();
+        
+        ArgumentCaptor<Request> capturedRequest = ArgumentCaptor.forClass(Request.class);
+        QueryResult result = queryClient.execute(funnel);
+        assertTrue(result instanceof FunnelResult);
+        FunnelResult funnelResult = (FunnelResult)result;
+
+        verify(mockHttpHandler).execute(capturedRequest.capture());
+        Request request = capturedRequest.getValue();
+
+        ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+        request.body.writeTo(outputStream);
+        String requestBody = outputStream.toString(ENCODING);
+        assertEquals(
+            "Unexpected request body",
+            "{\"steps\":[{\"timeframe\":\"this_7_days\",\"actor_property\":\"visitor.guid\",\"event_collection\":\"signed up\"},"
+          + "{\"timeframe\":\"this_7_days\",\"actor_property\":\"user.guid\",\"event_collection\":\"completed profile\"},"
+          + "{\"timeframe\":\"this_7_days\",\"timezone\":\"UTC\",\"actor_property\":\"user.guid\",\"event_collection\":\"referred user\"}]}",
+            requestBody);
```

Comment:
As discussed with Justin, I'd like to make sure this doesn't depend on the ordering of parameters that don't need to be ordered. Same goes for all other Funnel tests.

testAbsoluteTimeframe() also currently depends on the order of some unordered keys, so that should be updated as well.